pkg/sourcer/message.go

```diff
@@ -43,11 +43,11 @@ func (m Message) EventTime() time.Time {
 
 type Offset struct {
 	value       []byte
-	partitionId string
+	partitionId int32
```

Comment:
I was between `string` and `int` when I worked on the SDK. The question is "Are all user-defined sources using integers to represent partitions?" If the answer is yes, `int32` is perfect. If no, `string` would make more sense because either string or integer can be easily converted to string. If partitions are represented in string format, converting to a unique integer might require some conversion algorithms.

That's also why I named it partitonId instead of partitionIndex.

If we are sure there is no string representation for partitions. Let's go with int32. Otherwise, consider keeping it as string.

Reply:
if the `partitionId` is not a valid number, we are using `defaultPartitionId` (0) in Numaflow.

Reply:
Synced offline. if there is a string representation of a partition, let's give the users full control of how they want to hash it to a unique integer.